documentation/modules/auxiliary/gather/http/cve_2021_27850_apache_tapestry_hmac_key.md

```diff
@@ -0,0 +1,28 @@
+## Description
+
+This module exploits CVE-2021-27850 which affects several versions of Apache Tapestry. The versions concerned are at least 5.4.5, 5.5.0, 5.6.2 and 5.7.0. This CVE allows an attacker to leak the source code of a Tapestry server by requesting a particular `.class` file with its name through the url with an extra `/` at the end: http://tapestryhost:8080/assets/something/services/AppModule.class/. Due to this, it is possible to leak the HMAC secret key which is located in `AppModule.class` by default.
+
```

Comment:
Could you add some installation instructions for the vulnerable software or some links to downloadable software / installation instructions? The beginning of this section could look something like below:

```
### Installation

A vulnerable version of Apache Tapestry can be downloaded from [here](https://downloads.apache.org/tapestry/apache-tapestry-5.7.0-bin.zip) 
```

Reply:
I've added some information :)